src/System.Management.Automation/engine/TypeTable.cs

```diff
@@ -4489,6 +4489,13 @@ private bool ProcessIsBuiltIn(string filePath, ConcurrentBag<string> errors, out
                 ProcessTypeData(filePath, errors, Types_Ps1Xml.Get());
                 result = true;
             }
+            else if (string.Equals(Path.Combine(psHome, "typesv3.ps1xml"), filePath, StringComparison.OrdinalIgnoreCase))
+            {
+                // Workaround for a scenario where we get the 'typesv3.ps1xml' file name from public API.
+                // Now all types from 'typesv3.ps1xml' is in 'types.ps1xml' so we load 'types.ps1xml'.
+                ProcessTypeData("types.ps1xml", errors, Types_Ps1Xml.Get());
```

Comment:
This means `types.ps1xml` will be processed twice, which could potentially cause issue that I don't know of (conflicts?).

Reply:
There GetOrAdd() is used. (But it will slow down)

I'd prefer a breaking change (gray area) and get rid typesv3.ps1xml (and then types.ps1xml as James asked). It looks like very rarely edge case.
GitHub search https://github.com/search?q=TypeTable+typesv3.ps1xml&type=Code
This is certainly not an exhaustive test, but it returned nothing excluding PowerShell code.

Reply:
> This open way to make the code faster (currently ~30% in runspace creation).

I don't think merging those two source files and eliminating `types3.ps1xml` would help much on the perf here.
I will let the committee review the proposed change.

Reply:
> I don't think merging those two source files and eliminating types3.ps1xml would help much on the perf here.

My thought is that it opens a way for future optimizations.

Reply:
Just took a look at the code in `TypeTable.cs`, and duplicate TypeData will result in error during processing:
https://github.com/PowerShell/PowerShell/blob/36bc894d9be1f9a8274392375bd1079cfbb2f77b/src/System.Management.Automation/engine/TypeTable.cs#L2698-L2702

Reply:
I tried the compiled artifact and get the error. If PowerShell Committee conclusion will be to have a workaround I will discover another workaround but I hope we will agree to remove.